go/test/endtoend/onlineddl/scheduler/onlineddl_scheduler_test.go

```diff
@@ -991,7 +1053,679 @@ func TestSingleton(t *testing.T) {
+		qr, err := conn.ExecuteFetch(query, 1000, true)
+
+		func() {
+			writeMetrics.mu.Lock()
+			defer writeMetrics.mu.Unlock()
+
+			writeMetrics.deletesAttempts++
+			if err != nil {
+				writeMetrics.deletesFailures++
+				return
+			}
+			assert.Less(t, qr.RowsAffected, uint64(2))
+			if qr.RowsAffected == 0 {
+				writeMetrics.deletesNoops++
+				return
+			}
+			writeMetrics.deletes++
+		}()
+		return err
+	}
+
+	initTable := func(t *testing.T) {
+		log.Infof("initTable begin")
+		defer log.Infof("initTable complete")
+
+		ctx := context.Background()
+		conn, err := mysql.Connect(ctx, &vtParams)
+		require.Nil(t, err)
+		defer conn.Close()
+
+		writeMetrics.Clear()
+		_, err = conn.ExecuteFetch(truncateStatement, 1000, true)
+		require.Nil(t, err)
+
+		for i := 0; i < maxTableRows/2; i++ {
+			generateInsert(t, conn)
+		}
+		for i := 0; i < maxTableRows/4; i++ {
+			generateUpdate(t, conn)
+		}
+		for i := 0; i < maxTableRows/4; i++ {
+			generateDelete(t, conn)
+		}
+	}
+
+	testSelectTableMetrics := func(t *testing.T) {
+		writeMetrics.mu.Lock()
+		defer writeMetrics.mu.Unlock()
+
+		log.Infof("%s", writeMetrics.String())
+
+		ctx := context.Background()
+		conn, err := mysql.Connect(ctx, &vtParams)
+		require.Nil(t, err)
+		defer conn.Close()
+
+		rs, err := conn.ExecuteFetch(selectCountRowsStatement, 1000, true)
+		require.Nil(t, err)
+
+		row := rs.Named().Row()
+		require.NotNil(t, row)
+		log.Infof("testSelectTableMetrics, row: %v", row)
+		numRows := row.AsInt64("num_rows", 0)
+		sumUpdates := row.AsInt64("sum_updates", 0)
+
+		assert.NotZero(t, numRows)
+		assert.NotZero(t, sumUpdates)
+		assert.NotZero(t, writeMetrics.inserts)
+		assert.NotZero(t, writeMetrics.deletes)
+		assert.NotZero(t, writeMetrics.updates)
+		assert.Equal(t, writeMetrics.inserts-writeMetrics.deletes, numRows)
+		assert.Equal(t, writeMetrics.updates-writeMetrics.deletes, sumUpdates) // because we DELETE WHERE updates=1
+	}
+
+	testOnlineDDL := func(t *testing.T, alterStatement string, ddlStrategy string, executeStrategy string, expectHint string, expectError string) (uuid string) {
+		params := &testOnlineDDLStatementParams{
+			ddlStatement:    alterStatement,
+			ddlStrategy:     ddlStrategy,
+			executeStrategy: executeStrategy,
+			expectHint:      expectHint,
+			expectError:     expectError,
+		}
+		if executeStrategy != "vtgate" {
+			params.migrationContext = migrationContext
+		}
+		return testOnlineDDLStatement(t, params)
+	}
+	createRevertParams := func(revertUUID string) *testRevertMigrationParams {
+		return &testRevertMigrationParams{
+			revertUUID:      revertUUID,
+			executeStrategy: "vtctl",
+			ddlStrategy:     string(schema.DDLStrategyOnline),
+		}
+	}
+
+	// init-cleaup
+	t.Run("init: drop table", func(t *testing.T) {
+		// IF EXISTS is not supported in -declarative
+		uuid := testOnlineDDL(t, dropIfExistsStatement, "online", "vtgate", "", "")
+		onlineddl.CheckMigrationStatus(t, &vtParams, shards, uuid, schema.OnlineDDLStatusComplete)
```

Comment:
I'm curious why we have this and the following subtest here since they don't use `declarative`? IMO we should remove them as we're not testing the feature and we DO have subtests that confirm they fail.

Same holds true for all of the subtests which aren't using `declarativeStrategy`. Perhaps there's good reason to have them here though, which is fine. I do see that you had a comment about this for some towards the end (they're not declarative but fit best in this test).

Reply:
This particular execution ensures the table does not exist before we begin the actual `declarative` testing. In `onlineddl_declarative` this was not required, since we started with an empty cluster. But now that we have converged multiple tests into a single workflow, we need to perform pre-test cleanup. Specifically, the particular table actually does exist from a previous test.
I'm not using `-declarative` in this `DROP TABLE` because I want a "trusted" method before embarking on `-declarative` validations.